File: Data Structures & Algorithms/majority-element/submission-1.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = 1
        key = nums[0]
        for i in range(1,len(nums)):
            if key == nums[i]:
                count += 1
            else:
                count -= 1
            if count == 0:
                key = nums[i]
                count = 1
        return key

            

File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> h = new HashMap<>();
        for (int i=0; i < nums.length; i++){
            int comp = target - nums[i];
            if (h.containsKey(comp)){
                return new int [] {h.get(comp),i};
            }
            h.put(nums[i], i);
        }
    throw new IllegalArgumentException("No two sum solution");   
        
    }
}
